### PR TITLE
buy one get one more cppcheck fix

### PR DESCRIPTION
**Pull request by cujomalainey**

### CODE REVIEW

File: tools/tplg_parser/tplg_parser.c [hunk 1]
```diff
@@ -555,7 +555,7 @@ int tplg_load_one_control(struct snd_soc_tplg_ctl_hdr **ctl, char **priv_data,
 	case SND_SOC_TPLG_DAPM_CTL_VOLSW:
 		/* load mixer type control */
 		rewind_size = sizeof(struct snd_soc_tplg_ctl_hdr);
-		if (fseek(file, rewind_size * -1, SEEK_CUR)) {
+		if (fseek(file, (long)rewind_size * -1, SEEK_CUR)) {
```
Comment (plbossart) [inline tools/tplg_parser/tplg_parser.c:558]: are size_t and long always the same size?

**Review by plbossart — APPROVED** [review]

File: tools/tplg_parser/tplg_parser.c (hunk 1 above)
Comment (cujomalainey) [reply tools/tplg_parser/tplg_parser.c:558]: size_t is unsigned since negative size only makes sense in a black hole :)

Comment (plbossart) [reply tools/tplg_parser/tplg_parser.c:558]: i never found a definitive answer on this from Google searches.

**Comment by lgirdwood** [comment]
Looks like testbench needs some more updates
```
2740766d5 5447a4f7b 496b27ba8 (HEAD, pull/4071/merge) Merge 496b27ba83f537678fea943ca18db25745f2aad4 into 5447a4f7b1f19500664b546cee98c78d1d8ba15d
Scanning dependencies of target sof_tplg_parser
[ 50%] Building C object CMakeFiles/sof_tplg_parser.dir/tplg_parser.c.o
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c: In function 'tplg_load_pipeline':
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:507:9: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   array = (uint8_t *)array + array->size;
         ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:511:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  array = (uint8_t *)array - total_array_size;
        ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c: In function 'tplg_load_src':
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:890:9: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   array = (uint8_t *)array + array->size;
         ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:894:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  array = (uint8_t *)array - total_array_size;
        ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c: In function 'tplg_load_asrc':
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:970:9: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   array = (uint8_t *)array + array->size;
         ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:974:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  array = (uint8_t *)array - total_array_size;
        ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c: In function 'tplg_load_process':
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:1065:9: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   array = (uint8_t *)array + array->size;
         ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:1069:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  array = (uint8_t *)array - total_array_size;
        ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c: In function 'tplg_load_mixer':
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:1135:9: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
   array = (uint8_t *)array + array->size;
         ^
/home/sof/work/sof.git/tools/tplg_parser/tplg_parser.c:1139:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  array = (uint8_t *)array - total_array_size;
        ^
-- GIT_TAG / GIT_LOG_HASH : v1.7-rc1-277-g2740766d59c0 / 2740766d5
```

**Comment by cujomalainey** [comment]
> Looks like testbench needs some more updates
> 

Thanks, was waiting for CI to test this, fixed now

**Review by iuliana-prodan — APPROVED** [review]

File: tools/tplg_parser/tplg_parser.c (hunk 1 above)
Comment (cujomalainey) [reply tools/tplg_parser/tplg_parser.c:560]: seems like it can be `ULL`  on 64 bit and `U` on 32-bit, so 🤷

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**